### PR TITLE
New config option deploy.transformableAllowList

### PR DESCRIPTION
**Pull request by shawnzhu**
Supports: #4081 
**Related**: #6236
**Merge before/after**: _Dependent or prerequisite PRs_

**Description**
Add config option to configure a `transformableAllowList` so that it can transform CRDs. see [design proposal](https://github.com/GoogleContainerTools/skaffold/blob/1b2c994a68138bdac1d1967e82c310f92d36b588/docs/design_proposals/configurable-transformable-allowlist.md).

**User facing changes (remove if N/A)**

Before: the `transformableAllowList` is hard coded. it can only transform a limited number of resource types.
After: the `transformableAllowList` is configurable in `skaffold.yaml`. So that it can transform CRDs.

**Follow-up Work (remove if N/A)**

### CODE REVIEW

**Comment by shawnzhu** [comment]
@tejal29 may I ask for some guidance about this PR to see if I'm on the right track?

File: pkg/skaffold/schema/latest/v1/config.go [hunk 1]
```diff
@@ -517,6 +517,9 @@ type DeployConfig struct {
 
 	// Logs configures how container logs are printed as a result of a deployment.
 	Logs LogsConfig `yaml:"logs,omitempty"`
+
+	// TransformableAllowList configures an allowlist for transforming manifests.
+	TransformableAllowList []ResourceFilter `yaml:"transformableAllowList,omitempty"`
```
Comment (nkubala) [inline pkg/skaffold/schema/latest/v1/config.go:522]: let's hide this config field until it's used in code

```suggestion
	TransformableAllowList []ResourceFilter `yaml:"-,omitempty"`
```

**Review by nkubala — COMMENTED** [review]
@shawnzhu this looks solid! looks like as per https://github.com/GoogleContainerTools/skaffold/pull/6236 this is step 1 of the implementation plan. my only comment is I think we should hide the config option until it's used in code, otherwise this LGTM

**Review by aaron-prindle — APPROVED** [review]
LGTM.  Seems there are some test errors still though:
```
/pkg/skaffold/initializer/analyze/TestConfigAnalyzer/skaffold_config_equals_target_config
    --- FAIL: TestConfigAnalyzer/skaffold_config_equals_target_config (0.00s)
panic: duplicated key '-' in struct v1.DeployConfig [recovered]
	panic: duplicated key '-' in struct v1.DeployConfig [recovered]
	panic: duplicated key '-' in struct v1.DeployConfig
```

Remaining TODOs include:
- hiding the config option until it's used in code
- fixing unit tests

**Comment by shawnzhu** [comment]
I just saw `skaffold/v2beta22` has been released 2 days ago.

Should I rebase on `skaffold/v3alpha2` or make or wait for `skaffold/v2beta23`?

**Comment by shawnzhu** [comment]
rebasing `skaffold/v2beta23`

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
@shawnzhu Sorry for the delay. Approved. Triggering kokoro run and hopefully this will be in next release.

**Comment by shawnzhu** [comment]
@tejal29 no problem. I was trying to bump the code coverage a little bit beyond 70.41% to give this PR a breath

**Comment by annismckenzie** [comment]
This doesn't work yet, right? The field is marked with `yarm:"-"` so it's not usable in `skaffold.yaml` files. The v1.32.0 also mentions this as a feature in the release notes but it's not documented anywhere and it also can't work if the `yaml` annotation is set to `-`.

Am I missing something? 👀

**Comment by shawnzhu** [comment]
@annismckenzie you're right, it's still a work in progress so it's hidden. see https://github.com/GoogleContainerTools/skaffold/pull/6452#pullrequestreview-745394053

**Pull request merged**